Add scripts to configure GitHub repos

tijmenb opened the pull request:
This adds scripts to let us configure all of GOV.UK's repos.

It can:

- Set up the correct settings (only use merge commits)
- Set up the correct branch protection (for master, require PR reviews)
- Set up the correct webhooks to the GitHub Trello Poster

There's also a task we can use to clean up webhooks, like we have for Snyk.

https://github.com/alphagov/govuk-puppet/pull/6621 adds a Jenkins job that makes this script run every day.

boffbowsh commented on github/Rakefile:
```diff
@@ -0,0 +1,84 @@
+
+desc "Fetch the names of all GOV.UK repos from alphagov"
+task :fetch_repos do
+  repos = client.search_repositories("topic:govuk user:alphagov", per_page: 100).items.map(&:full_name)
```
This will only expose public repositories. If you want to do private ones too you'll need to use `org_repos` and filter by topic.

tijmenb replied:
👍 updated.

boffbowsh commented on github/Rakefile:
```diff
@@ -0,0 +1,84 @@
+desc "Remove old webhooks"
+task :remove_old_webhooks do
+  HOOKS_TO_DELETE = %w[
+    https://snyk.io/webhook/github
```
Yaaaass

boffbowsh commented on github/Rakefile:
```diff
@@ -0,0 +1,84 @@
+end
+
+desc "Create the correct webhooks for all repos"
+task :configure_webhooks do
```
Reckon this should also add the Jenkins build hook too?

tijmenb replied:
Yes, I'd like to do that in a separate PR though (I haven't figured out how to install an "Integration" web hook and I want to make sure this works first).

tijmenb commented:
@boffbowsh I've rewritten the PR.

- To automate the process, we now request the repos from GitHub without the intermediate writing to a YAML file
- Added tests

boffbowsh commented on README.md:
```diff
@@ -2,6 +2,10 @@
 Collection of scripts to configure Software as a Service applications we use.
 
-## Sentry
+## [Sentry][/sentry]
```
Wrong link syntax

tijmenb replied:
Fixed
